### ISSUE TITLE
Add support for private key JWT authentication as well as client ID and secret

### ISSUE THREAD

**Issue by aaronctmr** [auth0/auth0-deploy-cli#813]
### Checklist

- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Describe the problem you'd like to have solved

Along with providing the client ID and secret authentication mechanism, this library also supported private key JWT. In terms of a change, it seems relatively simple (see additional context). 

In the auth0-deploy-cli configuration file only the following options are currently supported in regard to client authentication:
```
"AUTH0_CLIENT_ID": "<client ID string>",
"AUTH0_CLIENT_SECRET": "<client secret string>",
```

### Describe the ideal solution

The following configuration is also supported:
```
    "AUTH0_CLIENT_SIGNING_KEY": "-----BEGIN RSA PRIVATE KEY-----\n< Contents of private key>\n-----END RSA PRIVATE KEY-----",
    "AUTH0_CLIENT_SIGNING_ALGORITHM": "RS256" /* optional */,
```

### Alternatives and current workarounds

N.A.

### Additional context

Sample patch for a working PoC of the desired change:

```
Index: node_modules/auth0-deploy-cli/lib/context/index.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/node_modules/auth0-deploy-cli/lib/context/index.js b/node_modules/auth0-deploy-cli/lib/context/index.js
--- a/node_modules/auth0-deploy-cli/lib/context/index.js	
+++ b/node_modules/auth0-deploy-cli/lib/context/index.js	
@@ -41,10 +41,10 @@
     if (!config.AUTH0_DOMAIN)
         missingParams.push('AUTH0_DOMAIN');
     if (!config.AUTH0_ACCESS_TOKEN) {
-        if (!config.AUTH0_CLIENT_ID)
-            missingParams.push('AUTH0_CLIENT_ID');
-        if (!config.AUTH0_CLIENT_SECRET)
-            missingParams.push('AUTH0_CLIENT_SECRET');
+      if (!config.AUTH0_CLIENT_ID)
+        missingParams.push('AUTH0_CLIENT_ID);
+      if (!config.AUTH0_CLIENT_SECRET && !config.AUTH0_CLIENT_SIGNING_KEY)
+        missingParams.push('AUTH0_CLIENT_SECRET or AUTH0_CLIENT_SIGNING_KEY');
     }
     if (missingParams.length > 0) {
         throw new Error(`The following parameters were missing. Please add them to your config.json or as an environment variable. ${JSON.stringify(missingParams)}`);
@@ -109,19 +109,30 @@
         if (!!config.AUTH0_ACCESS_TOKEN)
             return config.AUTH0_ACCESS_TOKEN;
 
-        const authClientConfig = {
+
+        let clientAuthOptions = {
           domain: config.AUTH0_DOMAIN,
           clientId: config.AUTH0_CLIENT_ID,
         };
 
-        if (config.AUTH0_)
+        if (!!config.AUTH0_CLIENT_SECRET) {
+          clientAuthOptions.clientSecret = config.AUTH0_CLIENT_SECRET;
+        } else if (!!config.AUTH0_CLIENT_SIGNING_KEY) {
+          clientAuthOptions.clientAssertionSigningKey = config.AUTH0_CLIENT_SIGNING_KEY;
+
+          if (!!config.AUTH0_CLIENT_SIGNING_ALGORITHM) {
+            clientAuthOptions.clientAssertionSigningAlg = config.AUTH0_CLIENT_SIGNING_ALGORITHM;
+          }
 
-        const authClient = new auth0_1.AuthenticationClient(authClientConfig);
+        }
+
+        const authClient = new auth0_1.AuthenticationClient(clientAuthOptions);
         const clientCredentials = yield authClient.clientCredentialsGrant({
             audience: config.AUTH0_AUDIENCE
                 ? config.AUTH0_AUDIENCE
                 : `https://${config.AUTH0_DOMAIN}/api/v2/`,
         });
+
         return clientCredentials.access_token;
     }))();
     const mgmtClient = new auth0_1.ManagementClient({
```

**Comment by aaronchilcott** [comment]
Looks like the upstream change #815 is causing the tests to fail.

**Comment by willvedd** [comment]
@aaronchilcott This has been released in [7.19.0](https://github.com/auth0/auth0-deploy-cli/releases/tag/v7.19.0). Thank you for your contribution!

**Comment by aaronchilcott** [comment]
Thanks @willvedd, my pleasure! Thanks for considering my changes =)
